server/src/routes/api.rs

```diff
@@ -4,119 +4,177 @@ use crate::api::community::*;
+          .wrap(rate_limit.post())
+          .route(web::post().to(route_post::<CreatePost>)),
+      )
+      .service(
+        web::scope("/post")
+          .wrap(rate_limit.message())
+          .route("", web::get().to(route_get::<GetPost>))
+          .route("", web::put().to(route_post::<EditPost>))
+          .route("/list", web::get().to(route_get::<GetPosts>))
+          .route("/like", web::post().to(route_post::<CreatePostLike>))
+          .route("/save", web::put().to(route_post::<SavePost>)),
+      )
+      // Comment
+      .service(
+        web::scope("/comment")
+          .wrap(rate_limit.message())
+          .route("", web::post().to(route_post::<CreateComment>))
+          .route("", web::put().to(route_post::<EditComment>))
+          .route("/like", web::post().to(route_post::<CreateCommentLike>))
+          .route("/save", web::put().to(route_post::<SaveComment>)),
+      )
+      // User
+      .service(
+        // Account action, I don't like that it's in /user maybe /accounts
+        // Handle /user/register separately to add the register() rate limitter
+        web::resource("/user/register")
+          .guard(guard::Post())
+          .wrap(rate_limit.register())
+          .route(web::post().to(route_post::<Register>)),
+      )
+      // User actions
+      .service(
+        web::scope("/user")
+          .wrap(rate_limit.message())
+          .route("", web::get().to(route_get::<GetUserDetails>))
+          .route("/mention", web::get().to(route_get::<GetUserMentions>))
+          .route("/mention", web::put().to(route_post::<EditUserMention>))
+          .route("/replies", web::get().to(route_get::<GetReplies>))
+          .route(
+            "/followed_communities",
+            web::get().to(route_get::<GetFollowedCommunities>),
+          )
+          // Admin action. I don't like that it's in /user
+          .route("/ban", web::post().to(route_post::<BanUser>))
+          // Account actions. I don't like that they're in /user maybe /accounts
+          .route("/login", web::post().to(route_post::<Login>))
+          .route(
+            "/delete_account",
+            web::post().to(route_post::<DeleteAccount>),
+          )
+          .route(
+            "/password_reset",
+            web::post().to(route_post::<PasswordReset>),
+          )
+          .route(
+            "/password_change",
+            web::post().to(route_post::<PasswordChange>),
+          )
+          // mark_all_as_read feels off being in this section as well
+          .route(
+            "/mark_all_as_read",
+            web::post().to(route_post::<MarkAllAsRead>),
+          )
+          .route(
+            "/save_user_settings",
+            web::put().to(route_post::<SaveUserSettings>),
+          ),
+      )
+      // Admin Actions
+      .service(
+        web::resource("/admin/add")
+          .wrap(rate_limit.message())
+          .route(web::post().to(route_post::<AddAdmin>)),
+      ),
+  );
```

Comment:
I didn't know you could scope layouts like this, nice. Its much clearer this way.